lib/cli.rb

```diff
@@ -23,71 +24,41 @@ def commands
   end
 
   def prompt
-    print '<Groove>'.cyan
+    print "<Groove>".cyan
   end
 
   def undefined
     puts "Command not found!!".red
     commands
   end
 
-  def list_songs
-    Song.all.each_with_index do |song, index|
-      puts "#{index + 1}. #{song.artist.name} - #{song.name}" \
+  def list_songs(song, index)
+    puts "#{index + 1}. #{song.artist.name} - #{song.name}" \
       " - #{song.genre.name}".light_blue
-    end
-  end
-
-  def list_artists
-    Artist.all.each { |artist| puts artist.name.to_s.light_blue }
   end
 
-  def list_genres
-    Genre.all.each { |genre| puts genre.name.to_s.light_blue }
+  def list_artists(artist)
+    puts artist.name.to_s.light_blue
   end
 
-  def songs_by_category(category)
-    if category
-      category.songs.each do |song|
-        puts "#{song.artist.name} - #{song.name}" \
-        " - #{song.genre.name}".light_blue
-      end
-    else
-      object_not_found(category.to_s)
-    end
+  def list_genres(genre)
+    puts genre.name.to_s.light_blue
   end
 
-  def play_song_for_real
-    @pid = fork { exec 'afplay', $song_path }
-  end
-
-  def stop_song
-    if process_running?
-      fork { exec 'killall afplay' }
-    else
-      puts "Play song first!".red
-    end
+  def songs_by_category(song)
+    puts "#{song.artist.name} - #{song.name} - #{song.genre.name}".light_blue
   end
 
   def playing_song(song)
-    if song
-      puts "Playing #{song.artist.name} - #{song.name}" \
+    puts "Playing #{song.artist.name} - #{song.name}" \
       " - #{song.genre.name}".light_blue
-    else
-      object_not_found("song")
-    end
-  end
-
-  def follow_up_prompt(song_object)
-    puts "Kindly Enter #{song_object}".yellow
   end
 
-  def object_not_found(song_object)
```

Comment:
song_object is does not describe the variable in this case. Use a more appropriate variable name